index.js

```diff
@@ -17,219 +24,240 @@ const TOKEN_PATH = 'token.json';
     return {
-      text: text,
-      name: name,
+      text,
+      name,
       replacement: EMOJI_BOOM,
-      unicode: EMOJI_BOOM
+      unicode: EMOJI_BOOM,
     };
   }
   return emoji;
-}
-
+};
 
 const react = async (message, emojis) => {
-  for (let emoji of emojis) {
-    if (typeof emoji.id != 'undefined') {
+  emojis.forEach(async (emoji) => {
+    if (typeof emoji.id !== 'undefined') {
       await message.react(emoji.id);
-    } else if (typeof emoji.unicode != 'undefined' ) {
+    } else if (typeof emoji.unicode !== 'undefined') {
       await message.react(emoji.unicode);
     } else {
       console.log(`Can't react with invalid emoji "${emoji.name}"`);
     }
+  });
+};
+
+const getMatchesCount = (round, size) => {
+  let a = 0;
+  let b = 0;
+  if (round === 4) {
+    a = 4;
+  } else if (round === 5) {
+    a = 2;
+  } else if (round === 6) {
+    a = 2;
+  } else {
+    a = 8;
   }
-}
-
-
-const getMatchesCount = (round, size) => Math.min(round === 4 ? 4 : (round === 5 ? 2 : (round === 6 ? 2 : 8)), Math.pow(2, round - (size > 64 ? 0 : (size > 32 ? 1 : 2))));
-
+  if (size > 64) {
+    b = 0;
+  } else if (size > 32) {
+    b = 1;
+  } else {
+    b = 2;
+  }
+  Math.min(a, 2 ** (round - b));
+};
 
-const nextMatch = async matches => {
+const nextMatch = async (matches) => {
   const channel = client.channels.cache.get(CHANNEL_ID);
 
-  let botState = await getValue(BOT_STATE_REF)
+  const botState = await getValue(BOT_STATE_REF);
 
-  if (!isBotEnabled(botState)) {
+  if (!isBotEnabled(botState) && !skipstat) {
     console.log(`${now}: Bot disabled, exiting`);
     return;
   }
 
   // Even though REFS is an object, order is guaranteed for non-string keys
-  let valueRanges = await getValues(Object.values(REFS));
-
-  let round = valueRanges[0].values[0].toString();
-  let song = parseInt(valueRanges[1].values[0].toString());
-  let header = ('values' in valueRanges[2]) ? valueRanges[2].values[0].toString() : null;
-  let footer = ('values' in valueRanges[3]) ? valueRanges[3].values[0].toString() : null;
-  let size = parseInt(valueRanges[5].values[0].toString());
-  let year = parseInt(valueRanges[6].values[0].toString());
-  let rndVal = parseInt(round.slice(1));
-
-  if (typeof matches == 'undefined') {
-    matches = getMatchesCount(rndVal, size);
-    console.log(`${now}: Posting ${matches} matches this iteration`)
+  const valueRanges = await getValues(Object.values(REFS));
+
+  const round = valueRanges[0].values[0].toString();
+  const song = parseInt(valueRanges[1].values[0].toString());
+  const header = ('values' in valueRanges[2]) ? valueRanges[2].values[0].toString() : null;
+  const footer = ('values' in valueRanges[3]) ? valueRanges[3].values[0].toString() : null;
+  const size = parseInt(valueRanges[5].values[0].toString());
+  const year = parseInt(valueRanges[6].values[0].toString());
+  const rndVal = parseInt(round.slice(1));
+
+  let matchCount = matches;
+  if (typeof matcheCount === 'undefined') {
```

Comment:
Typo here, `matcheCount` should be `matchCount`

Reply:
Good catch.